crates/omnix-cli/src/command/mod.rs

```diff
@@ -26,11 +26,33 @@ pub enum Command {
 
 impl Command {
     pub async fn run(&self, verbosity: Verbosity<InfoLevel>) -> anyhow::Result<()> {
+        let check_nix_installed_and_exit = || {
+            if omnix_common::check::nix_installed() {
+                return;
+            }
+            tracing::error!(
+                "Nix is not installed. Please install Nix: https://nixos.asia/en/install"
+            );
+            std::process::exit(1);
+        };
+
         match self {
-            Command::Show(config) => config.run().await,
-            Command::Init(config) => config.run().await,
-            Command::CI(config) => config.run(verbosity).await,
-            Command::Health(config) => config.run().await,
+            Command::Show(config) => {
+                check_nix_installed_and_exit();
+                config.run().await
+            }
```

Comment:
Just inline `check_nix_installed_and_exit` at the top and unconditionally evaluate it (unless `self == Command::Completion`). It is much more clear this way (don't need to do `O(n)` code scan on match arms to understand the logic).

Reply:
https://github.com/juspay/omnix/pull/213/commits/7881fdd9019544d3c78fcfcad71f792aca66eae0